transcript/src/main/java/com/transcriptanalyzer/transcript/User_Requests_Intents/Service/Tree.java

```diff
@@ -0,0 +1,108 @@
+    private Node treeRootNode;
+    private HashMap<String, Integer> counts;
+
+    // class for tree representing intents as it's nodes
+    // intents organized in order of their turn in their respective transcript
+    public Tree(ArrayList<ArrayList<ArrayList<String>>> transcripts) {
+        // root of tree
+        this.treeRootNode = new Node("Root", null);
+        Node start = this.treeRootNode;
+        this.counts = new HashMap<String, Integer>();
+
+        // go through each transcript, then each turn (representing each message intent pair)
+        // and add it as a subIntent node of the treeNode, then update treeNode down
+        for (ArrayList<ArrayList<String>> transcript : transcripts) {
+            for (ArrayList<String> turn : transcript) {
+                if (turn.size() == 2) {
+                    String currIntent = turn.get(1);
+                    // dont create node if it's none
+                    if (!Objects.equals(currIntent, "None")) {
+                        Node newRoot = this.treeRootNode.addSubIntent(currIntent);
+                        // add intent if doesnt exist, increment existing count if exists
+                        this.counts.merge(newRoot.intent, 1, Integer::sum);
+                        this.treeRootNode = newRoot;
+                    }
+                }
+            }
+            // at the end of each transcript bring treeNode back to top
+            this.treeRootNode = start;
+        }
+    }
+
+    // class for each individual intent
+    public class Node {
+        private String intent;
+        // # of number it has appeared in its respective turn
+        private int occurrences;
+        private Node parent;
+        private ArrayList<Node> children;
+
+        public Node(String intent, Node parent){
+            this.intent = intent;
+            this.occurrences = 1;
+            this.parent = parent;
+            this.children = new ArrayList<Node>();
+        }
+
+        public Node addSubIntent(String intent) {
+            // if the intent already exists as a child of the parent node
+            if (this.hasChild(intent) != null) {
+                Node node = this.hasChild(intent);
+                // increment occurrences of this node
+                node.occurrences += 1;
+                return node;
+                // if the intent is not a child of the node
+            } else {
+                // create the new node and add it as a child of the parent node
+                Node newChildNode = new Node(intent, this);
+                this.children.add(newChildNode);
+                return newChildNode;
+            }
+        }
+
+        // return subNode if it matches intent, o.w return None
+        public Node hasChild(String intent){
+            for (Node child : this.children) {
+                if (Objects.equals(intent, child.intent)) {
+                    return child;
+                }
+            }
+            return null;
+        }
+    }
```

Comment:
Instead of having an explicit return of null, consider making the return type Optional<Node>.